src/arrayMethodSort.js

```diff
@@ -4,8 +4,37 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      }
+
+      if (String(a) > String(b)) {
+        return 1;
+      }
+
+      return 0;
+    },
+  ) {
+    let swapped;
+
+    for (let i = 0; i < this.length; i++) {
+      swapped = false;
+
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          [this[i], this[j]] = [this[j], this[i]];
+          swapped = true;
+        }
+      }
+
+      if (!swapped) {
+        break;
+      }
```

Comment:
The 'swapped' flag is used to detect if any elements were swapped during the inner loop, allowing for an early break if the array is already sorted. This is a good optimization for the bubble sort algorithm.